javascript/javascript2/week3/fastest_presser/main.js

```diff
@@ -0,0 +1,90 @@
+let gameDuration;
+const inputDuration = document.querySelector("input");
+const startBtn = document.querySelector(".start-game");
+const resetBtn = document.querySelector(".reset-game");
+const sSpan = document.getElementById("s-amount");
+const jSpan = document.getElementById("j-amount");
+const sBlock = document.getElementById("s-block");
+const jBlock = document.getElementById("j-block");
+const gameMessage = document.querySelector(".game-message");
+const gameTimer = document.getElementById("game-timer");
+
+let sCounter = 0;
+let jCounter = 0;
+
+function countKeyDown(key) {
+  if (key.code === "KeyS") {
+    sCounter += 1;
+    sSpan.innerText = sCounter;
+  }
+  if (key.code === "KeyJ") {
+    jCounter += 1;
+    jSpan.innerText = jCounter;
+  }
+}
+function setCountDown(time) {
+  gameTimer.innerText = time;
+  let timeleft = time - 1;
+  const downloadTimer = setInterval(function () {
+    if (timeleft <= 0) {
+      clearInterval(downloadTimer);
+    }
+    gameTimer.innerText = timeleft;
+    timeleft -= 1;
+  }, 1000);
+}
+startBtn.addEventListener("click", () => {
+  gameDuration = parseInt(inputDuration.value);
+  if (!isNaN(gameDuration)) {
+    setCountDown(gameDuration);
+    startBtn.disabled = true;
+    setTimeout(() => {
+      window.removeEventListener("keydown", countKeyDown);
```

Comment:
In here you would probably want to use "keyup" event, as "keydown" allows players to cheat if they keep pressing the key. Keydown fires the event while the key is down it keeps running the event again and again, while keyup fires the event on key release.